runtime/client_http_response_test.go

```diff
@@ -230,6 +242,79 @@ func TestUnknownStatusCode(t *testing.T) {
 	assert.Len(t, logs["Finished an outgoing client HTTP request"], 1)
 }
 
+func TestNotFollowRedirect(t *testing.T) {
+	gateway, err := benchGateway.CreateGateway(
+		defaultTestConfig,
+		&testGateway.Options{
+			LogWhitelist: map[string]bool{
+				"Could not emit statusCode metric": true,
+			},
+			KnownHTTPBackends: []string{"bar"},
+			ConfigFiles:       util.DefaultConfigFiles("example-gateway"),
+		},
+		exampleGateway.CreateGateway,
+	)
+	if !assert.NoError(t, err) {
+		return
+	}
+	defer gateway.Close()
+
+	bgateway := gateway.(*benchGateway.BenchGateway)
+
+	addr := bgateway.HTTPBackends()["bar"].RealAddr
+	baseURL := "http://" + addr
+
+	redirectURI := baseURL + "/bar-path"
+	fakeEcho := func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Add("Location", redirectURI)
+		w.WriteHeader(303)
+		_, err := w.Write([]byte(`false`))
+		assert.NoError(t, err)
+	}
+
+	fakeNormal := func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(999)
+		_, err := w.Write([]byte(`false`))
+		assert.NoError(t, err)
+	}
+
+	bgateway.HTTPBackends()["bar"].HandleFunc("POST", "/bar/echo", fakeEcho)
+	bgateway.HTTPBackends()["bar"].HandleFunc("GET", "/bar-path", fakeNormal)
+
+	client := zanzibar.NewHTTPClientContext(
+		bgateway.ActualGateway.Logger,
+		bgateway.ActualGateway.ContextMetrics,
+		jsonwrapper.NewDefaultJSONWrapper(),
+		"bar",
+		map[string]string{
+			"echo": "bar::echo",
+		},
+		baseURL,
+		map[string]string{},
+		time.Second,
+		false,
```

Comment:
@rpatali does this test case address your ask? in this test case, it creates the gateway via exampleGateway and use example gateway's config files. And this test case is to verify the client call doesn't follow the redirect, so the caller will receive 303 status code.

Reply:
this is fine but additional also set the config value for `clients.*.followRedirects = false` for one of the clients in the <env>.json to test the behavior since that is what you are going to need.